sonarqube/datadog_checks/sonarqube/check.py

```diff
@@ -185,18 +264,36 @@ def compile_metric_patterns(config, field):
         return re.compile('|'.join(patterns)) if patterns else None
 
     @staticmethod
-    def create_metric_matcher(pattern, default):
+    def compile_component_patterns(config, field):
+        component_patterns = config.get(field, [])
+        if not isinstance(component_patterns, list):
+            raise ConfigurationError('The `{}` setting must be an array'.format(field))
+        patterns = []
+        for i, component_pattern in enumerate(component_patterns, 1):
+            if not isinstance(component_pattern, str):
+                raise ConfigurationError('Pattern #{} in `{}` setting must be a string'.format(i, field))
+            # Ensure dots are treated as literal
+            component_pattern = component_pattern.replace('\\.', '.').replace('.', '\\.')
+            # Get rid of any explicit start modifiers
+            component_pattern = component_pattern.lstrip('^')
+            # Match from the start by default
+            component_pattern = '^{}'.format(component_pattern)
```

Comment:
`Ensure dots are treated as literal` conflicts with `Match from the start by default` because then users can't do `.+substring`

We should remove the dot logic so it's pure regex like all other integrations, thus reducing confusion for users

Reply:
I agree with you but this code is previous to this PR and I would like to handle that in a different PR evaluating the consequences of changing this.